### DISCUSSION TITLE
Dream for new Media Files - View of audio / video + some additional properties

### DISCUSSION THREAD

**Discussion by JeffreyPerplex** [umbraco/Umbraco-CMS#10437] [Features and ideas]
I love the Media files that looks like first-citizens now (like audio + video)

![afbeelding](https://user-images.githubusercontent.com/12457148/121577050-63193b80-ca29-11eb-9c44-baf964cfb4ba.png)

How cool would it be if we could add some additional properties, but also the preview of the audio and video like we do with images at this moment. With type "Image" you get width + height out of the box. Would be great if we would have some specifics for audio / video as well

Image (at this moment):

![afbeelding](https://user-images.githubusercontent.com/12457148/121577150-85ab5480-ca29-11eb-9951-89e3680d7dfa.png)

Video (at this moment):

![afbeelding](https://user-images.githubusercontent.com/12457148/121577196-94920700-ca29-11eb-93a6-a55e88dcb33d.png)

Audio (at this moment):

![afbeelding](https://user-images.githubusercontent.com/12457148/121577217-9a87e800-ca29-11eb-92ef-c49d1898db12.png)

============

Cooler would be (totally ripped of this great presentation ;): https://www.perplex.nl/media/108667/i-have-a-dream-about-umbraco-8.pdf (slide 42):

For video
![afbeelding](https://user-images.githubusercontent.com/12457148/121577317-affd1200-ca29-11eb-8c74-141e67980991.png)

- Viewing the video directly
- Additional property: "Length" and "Resolution"?

For audio:

![afbeelding](https://user-images.githubusercontent.com/12457148/121577438-cefba400-ca29-11eb-915f-9c1c58ae4576.png)

- Play the audio directly
- Additional property: "Length" and "bitrate"

**Comment by RachBreeze** [comment]
This would be amazing especially if we could tie the videos and audio element with an accessibility update, and allow content authors to upload a WebVTT file. Thus allowing it to be used in the `track` element  https://www.w3.org/TR/webvtt1/  as has been documented here  for WordPress https://www.coywolf.news/webmaster/wordpress-video-captions-subtitles/

**Reply by nul800sebastiaan** [reply]
@RachBreeze to make it clear, that would be an additional upload for video and audio files right, so they have to live side-by-side on the same item?

**Reply by RachBreeze** [reply]
This is what I was thinking yes, and also for the YouTube embed picker

**Reply by RachBreeze** [reply]
Sorry to expand (and based on drupal issue, https://www.drupal.org/project/drupal/issues/3002770) as it came from the same accessibility cluster https://accessibilitycluster.com/ work

Because we  video and audio as a core media type we need to consider 

**Transcripts**
1. ATAG: Add a transcript field to local audio and video media bundles in standard profile. A long text, formatted, permitting HTML.
2. WCAG: Render the transcript field in manage display. Transcripts can be long, so it's probably worth using a HTML summary/details element to wrap this, and place it after the the audio/video element itself.
3. ATAG: add transcripts advice for authors 

*Support Captions, Subtitles (and other track kinds)*
1. ATAG: Add a Media Track field type. This will be a compound field type, comprising a file upload, and some additional properties for the HTML <track> attributes. For this MVP plan, we will only support WebVTT files, and assume authors know how to export (or convert) this format with their caption editor tools. The expectation is that authors will prepare tracks using a caption-editor application, then upload a WebVTT format text file. Bonus: it should be trivial to support the other <track> attribute values specified by the HTML5 recommendation. This just means ensuring that authors can specify all of the srclang, kind, label, default attributes. Browsers don't support all kind values yet (like audio descriptions and chapters) but some 3rd-party JS video players do.
2. WCAG: Render the captions using a HTML <track> element nested inside the <audio> and <video> elements. 
3. ATAG: add advice about track support for authors,

For reference this in the prototype the cluster designed (CMS agnostic)
The accessibility cluster https://accessibilitycluster.com/ also identified these elements for a video
![image](https://user-images.githubusercontent.com/10153922/121901935-c3a4c300-cd1e-11eb-85f7-ab19b6021b46.png)

**Comment by nul800sebastiaan** [comment]
We're dreaming of very similar things @JeffreyPerplex, thank you! We're working on better audio/video previews at least for one of the next few minors. 👍 

Other things here could be interesting to work on as well, we haven't decided on those yet. :)
